HospitalSystem.ConsoleUI/PatientUI.cs

```diff
@@ -1058,5 +1102,130 @@ public void SetNotificationSettings()
+        foreach( var question in survey.Questions)
+        {
+            System.Console.WriteLine(question);
+            string answer = ReadSanitizedLine();
+            answers.Add(answer);
+        }
+        return answers;
+    }
+
+     List<int?> AnswerRatingQuestions(Survey survey)
+    {
+        List<int?> ratings = new();
+        foreach( var ratingQuestion in survey.RatingQuestions)
+        {
+            System.Console.WriteLine(ratingQuestion);
+            int rating;
+            while (true)
+            {
+                System.Console.Write("Please enter a rating between 1 and 5: ");
+                try
+                {
+                    rating = ReadInt(1, 5);
+                    break;
+                }
+                catch (InvalidInputException e)
+                {
+                    System.Console.WriteLine(e.Message + " Please try again.");
+                }
+            }
+            ratings.Add(rating);
+        }
+        return ratings;
+    }
+
+     public void CompleteSurvey(Survey survey,Doctor? doctor = null)
+    {
+        List<string?> answers = AnswerQuestions(survey);
+        List<int?> ratings = AnswerRatingQuestions(survey);
+        SurveyResponse response = new(answers,ratings,_loggedInPatient.Id);
+
+        if (doctor is not null)
+        {
+            _hospital.DoctorSurveyService.AddResponse((DoctorSurvey)survey,response,(Doctor)doctor);
+            return;
+        }
+        _hospital.HospitalSurveyService.AddResponse((HospitalSurvey)survey,response);
+    }
+
+    public void RateHospital()
+    {
+       PrintHospitalSurveys();
+       var surveys = _hospital.HospitalSurveyService.GetAll().ToList();
+       if (surveys.Count == 0)
+       {
+            return;
+       }
+       System.Console.Write("Please enter a number from a list: ");
+       int selectedIndex;
+       try
+        {
+            selectedIndex = ReadInt(0, surveys.Count-1);
+        }
+        catch (InvalidInputException e)
+        {
+            System.Console.Write(e.Message + " Aborting...");
+            return;
+        }
+        CompleteSurvey(surveys[selectedIndex]);
+        System.Console.WriteLine("Survey completed.");
+    }
+
+    public void RateDoctor(Doctor doctor, Patient patient)
+    {
+        var surveys = _hospital.DoctorSurveyService.GetSpecificDoctorUnansweredBy(patient,doctor).ToList();
+        PrintDoctorSurveys(surveys);
+        if (surveys.Count == 0)
+        {
+                return;
+        }
+        System.Console.Write("Please enter a number from a list: ");
+        int selectedIndex;
+        try
+        {
+            selectedIndex = ReadInt(0, surveys.Count-1);
+        }
+        catch (InvalidInputException e)
+        {
+            System.Console.Write(e.Message + " Aborting...");
+            return;
+        }
+
+        CompleteSurvey(surveys[selectedIndex],doctor);
+        System.Console.WriteLine("Survey completed.");
+    }
```

Comment:
This file has 1231 lines

Reply:
When a file is this long, github does not even support syntax coloring anymore................

Reply:
yikes

Reply:
will try to break it down